torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -88,43 +91,43 @@ void fuseBroadcast(Block *b) {
     // Can't fuse into nodes that don't support broadcasting
     if (!isBroadcasting(n)) continue;
 
-    // If the node already broadcasts, can't "rebroadcast"
-    // TODO: Actually, maybe you can, if there is a broadcast for some
-    // dims, and then another broadcast for the rest.  But this will
-    // never happen in practice so I didn't implement it.
-    if (n->hasAttribute(attr::broadcast) && n->i(attr::broadcast)) continue;
     JIT_ASSERT(!n->hasAttribute(attr::axis));
 
-    auto input_index = n->inputs().size() - 1;
-    auto* rhs_expand = n->input(input_index)->node();
-
-    // The rhs_expand input isn't actually an expand, so no fusion available
-    // XXX: we can't use the ->matches(...) mechanism in here, because input nodes
-    //      have been
-    if (rhs_expand->kind() != aten::expand ||
-        rhs_expand->input(1)->node()->kind() != onnx::Constant ||
-        rhs_expand->input(2)->node()->kind() != onnx::Constant) {
-      continue;
+    size_t index_begin = n->inputs().size() - 2;
+    if (n->kind() == onnx::Gemm) {
+      index_begin = n->inputs().size() - 1;
     }
+    for (size_t input_index = index_begin; input_index < n->inputs().size(); ++input_index) {
+      auto* rhs_expand = n->input(input_index)->node();
 
-    auto* unexpanded_rhs = rhs_expand->input(0);
+      // The rhs_expand input isn't actually an expand, so no fusion available
+      // XXX: we can't use the ->matches(...) mechanism in here, because input nodes
+      //      have been
+      if (rhs_expand->kind() != aten::expand ||
+          rhs_expand->input(1)->node()->kind() != onnx::Constant ||
+          rhs_expand->input(2)->node()->kind() != onnx::Constant) {
+        continue;
+      }
 
-    // We need to know what the type pre-expand is.  We should basically
-    // always have this information (because expands are only ever traced,
-    // not generated from symbolic), but if for some reason we don't
-    // have it, we need to skip.
-    if (!unexpanded_rhs->isTensor()) continue;
+      auto* unexpanded_rhs = rhs_expand->input(0);
 
-    // Not all broadcasts are supported by ONNX broadcast.
-    at::optional<size_t> axis = fusibleExpandTo(
-        unexpanded_rhs->type()->expect<TensorType>()->sizes(), // from
-        rhs_expand->output()->type()->expect<TensorType>()->sizes()); // to
-    if (axis == at::nullopt)
-      continue;
+      // We need to know what the type pre-expand is.  We should basically
+      // always have this information (because expands are only ever traced,
```

Comment:
Don't you need to update this comment? You seem to now be asserting you need to know what the output type too is now.